File: 성민/레벨2/타겟_넘버.js
function solution(numbers, target) {
  let answer = 0;
  const arr = [[+numbers[0], -numbers[0]]];
  for (let i = 1; i < numbers.length; i++) {
    const tempArr = [];
    for (const number of arr[i - 1]) {
      tempArr.push(number + numbers[i]);
      tempArr.push(number - numbers[i]);
    }
    arr.push(tempArr);
  }
  arr[arr.length - 1].forEach((v) => {
    if (v === target) {
      answer++;
    }
  });

  return answer;
}

console.log(solution([1, 1, 1, 1, 1], 3));

File: 신현/programmers/level2/타겟넘버.py
def solution(numbers, target):
    from collections import deque
    answer = 0
    n_len = len(numbers)
    q = deque([  [0,numbers[0]] , [0,-numbers[0]] ])
    print(q)
    while q:
        idx, t_val = q.popleft()
        idx+=1
        print(idx)
        if(idx < n_len):
            q.append([idx, t_val + numbers[idx]])
            q.append([idx, t_val - numbers[idx]])
        else:
            if t_val == target:
                answer+=1
    return answer